src/libYARP_OS/src/ManagedBytes.cpp

```diff
@@ -179,3 +179,16 @@ bool ManagedBytes::write(ConnectionWriter& writer) {
     writer.convertTextMode();
     return !writer.isError();
 }
+
+void ManagedBytes::setExternal(const char *extData, size_t len)
+{
+    if (!extData)
+    {
+        return;
+    }
+    clear();
+    owned = false;
+    Bytes extb(const_cast<char*>(extData), len);
```

Comment:
I was wondering why setExternal accepts `const` and if this is safe...
The `Bytes b` is not `const`, therefore the `const char` can be modified later by the user by using `Vector` methods.

Reply:
This may be actually the trieckiest part of this PR. The `Vector` does not check if it owns the memory (with e.g. a `bool owned` like `ManagedBytes`) and in fact it should not do it. We must understand whether there is a risk of memory modification or not when the memory is not owned.

Reply:
This may be actually the trieckiest part of this PR. The `Vector` does not check if it owns the memory (with e.g. a `bool owned` like `ManagedBytes`) and in fact it should not do it. We must understand whether there is a risk of memory modification or not when the memory is not owned.